app/channel.go

```diff
@@ -1039,22 +1039,37 @@ func (a *App) postLeaveChannelMessage(user *model.User, channel *model.Channel)
 }
 
 func (a *App) PostAddToChannelMessage(user *model.User, addedUser *model.User, channel *model.Channel, postRootId string) *model.AppError {
-	post := &model.Post{
+	message := fmt.Sprintf(utils.T("api.channel.add_member.added"), addedUser.Username, user.Username)
+	props := model.StringInterface{
+		"username":      user.Username,
+		"addedUsername": addedUser.Username,
+	}
+
+	systemPost := &model.Post{
 		ChannelId: channel.Id,
-		Message:   fmt.Sprintf(utils.T("api.channel.add_member.added"), addedUser.Username, user.Username),
+		Message:   message,
 		Type:      model.POST_ADD_TO_CHANNEL,
 		UserId:    user.Id,
-		RootId:    postRootId,
-		Props: model.StringInterface{
-			"username":      user.Username,
-			"addedUsername": addedUser.Username,
-		},
+		Props:     props,
 	}
-
-	if _, err := a.CreatePost(post, channel, false); err != nil {
+	if _, err := a.CreatePost(systemPost, channel, false); err != nil {
 		return model.NewAppError("postAddToChannelMessage", "api.channel.post_user_add_remove_message_and_forget.error", nil, err.Error(), http.StatusInternalServerError)
 	}
 
+	// send ephemeral post when postRootId is present since it's from RHS
+	// where mentioned user link is triggered.
+	if postRootId != "" {
```

Comment:
That comment isn't exactly true. Having a root ID just means it's a reply, doesn't necessarily mean the add is being triggered from the RHS, you can see and click on replies in the centre channel. The server has no real concept of the RHS, that's an entirely client-side construct

I'm a little wary of accepting this because it's very obviously server functionality to fit our exact clients. For a different client this might be really weird behavior, we're creating an added to channel post but also an ephemeral message saying the same thing? Those clients needs to add a special exception to handle this now or they will have two messages display every time this happens

It would be much better in my opinion if we handled this client-side on the web app, 3/5